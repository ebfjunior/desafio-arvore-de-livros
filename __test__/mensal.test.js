import React from 'react';

import Anual from '../src/components/anual';

describe('Componente Anual', () => {
  it('Deve renderizar adequadamente', () => {
    const periodos = {"2016-10-12" : 1, "1978-10-10" : 14,"2016-12-03" : 38};
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2015-12-03'} data_inicial={'2016-12-03'}/>);

    expect(wrapper).toMatchSnapshot();
  });

  it('State PerÃ­odos deve ser inicializado corretamente', () => {
    const periodos = {"2016-10-12" : 1, "1978-10-10" : 14,"2016-12-03" : 38};
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2015-12-03'} data_final={'2016-12-03'}/>);

    expect(Object.keys(wrapper.state().periodos).length).toEqual(3);
  });

  it('State Data Final deve ser inicializado corretamente', () => {
    const periodos = {"2016-10-12" : 1, "1978-10-10" : 14,"2016-12-03" : 38};
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2015-12-03'} data_final={'2016-12-03'}/>);

    expect(wrapper.state().data_final).toEqual('2016-12-03');
  });

  it('State Data Inicial deve ser um ano antes que a Data Final', () => {
    const periodos = {"2016-10-12" : 1, "1978-10-10" : 14,"2016-12-03" : 38};
    const wrapper = mount(<Anual periodos={periodos} data_inicial={'2015-12-03'} data_final={'2016-12-03'}/>);

    expect(wrapper.state().data_inicial).toEqual('2015-12-03');
  });
});