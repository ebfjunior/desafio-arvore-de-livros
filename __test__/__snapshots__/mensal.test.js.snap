// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Mensal Deve renderizar adequadamente 1`] = `
<Mensal
  ano="2016"
  mes="10"
  periodosMes={
    Object {
      "3": Object {
        "2016-10-12": 1,
      },
      "4": Object {
        "2016-10-23": 38,
      },
    }
  }
>
  <div
    className="mensal"
  >
    <Semanal
      ano="2016"
      dias={
        Array [
          null,
          null,
          null,
          null,
          null,
          null,
          1,
        ]
      }
      key="2016101"
      mes="10"
      periodosSemana={Object {}}
      semana={1}
    >
      <div
        className="semanal"
      >
        <Diario
          date={null}
          key="2016100.7567429870832711"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date={null}
          key="2016100.09869354567490518"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date={null}
          key="2016100.383161497535184"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date={null}
          key="2016100.18908561998978257"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date={null}
          key="2016100.067355529172346"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date={null}
          key="2016100.4139016654808074"
          produtividade={null}
        >
          <div
            className="diario cnull inactive"
            title={null}
          />
        </Diario>
        <Diario
          date="2016-10-01"
          key="2016101"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-01"
          />
        </Diario>
      </div>
    </Semanal>
    <Semanal
      ano="2016"
      dias={
        Array [
          2,
          3,
          4,
          5,
          6,
          7,
          8,
        ]
      }
      key="2016102"
      mes="10"
      periodosSemana={Object {}}
      semana={2}
    >
      <div
        className="semanal"
      >
        <Diario
          date="2016-10-02"
          key="2016102"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-02"
          />
        </Diario>
        <Diario
          date="2016-10-03"
          key="2016103"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-03"
          />
        </Diario>
        <Diario
          date="2016-10-04"
          key="2016104"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-04"
          />
        </Diario>
        <Diario
          date="2016-10-05"
          key="2016105"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-05"
          />
        </Diario>
        <Diario
          date="2016-10-06"
          key="2016106"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-06"
          />
        </Diario>
        <Diario
          date="2016-10-07"
          key="2016107"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-07"
          />
        </Diario>
        <Diario
          date="2016-10-08"
          key="2016108"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-08"
          />
        </Diario>
      </div>
    </Semanal>
    <Semanal
      ano="2016"
      dias={
        Array [
          9,
          10,
          11,
          12,
          13,
          14,
          15,
        ]
      }
      key="2016103"
      mes="10"
      periodosSemana={
        Object {
          "2016-10-12": 1,
        }
      }
      semana={3}
    >
      <div
        className="semanal"
      >
        <Diario
          date="2016-10-09"
          key="2016109"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-09"
          />
        </Diario>
        <Diario
          date="2016-10-10"
          key="20161010"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-10"
          />
        </Diario>
        <Diario
          date="2016-10-11"
          key="20161011"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-11"
          />
        </Diario>
        <Diario
          date="2016-10-12"
          key="20161012"
          produtividade={1}
        >
          <div
            className="diario c1 "
            title="2016-10-12"
          />
        </Diario>
        <Diario
          date="2016-10-13"
          key="20161013"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-13"
          />
        </Diario>
        <Diario
          date="2016-10-14"
          key="20161014"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-14"
          />
        </Diario>
        <Diario
          date="2016-10-15"
          key="20161015"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-15"
          />
        </Diario>
      </div>
    </Semanal>
    <Semanal
      ano="2016"
      dias={
        Array [
          16,
          17,
          18,
          19,
          20,
          21,
          22,
        ]
      }
      key="2016104"
      mes="10"
      periodosSemana={
        Object {
          "2016-10-23": 38,
        }
      }
      semana={4}
    >
      <div
        className="semanal"
      >
        <Diario
          date="2016-10-16"
          key="20161016"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-16"
          />
        </Diario>
        <Diario
          date="2016-10-17"
          key="20161017"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-17"
          />
        </Diario>
        <Diario
          date="2016-10-18"
          key="20161018"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-18"
          />
        </Diario>
        <Diario
          date="2016-10-19"
          key="20161019"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-19"
          />
        </Diario>
        <Diario
          date="2016-10-20"
          key="20161020"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-20"
          />
        </Diario>
        <Diario
          date="2016-10-21"
          key="20161021"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-21"
          />
        </Diario>
        <Diario
          date="2016-10-22"
          key="20161022"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-22"
          />
        </Diario>
      </div>
    </Semanal>
    <Semanal
      ano="2016"
      dias={
        Array [
          23,
          24,
          25,
          26,
          27,
          28,
          29,
        ]
      }
      key="2016105"
      mes="10"
      periodosSemana={Object {}}
      semana={5}
    >
      <div
        className="semanal"
      >
        <Diario
          date="2016-10-23"
          key="20161023"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-23"
          />
        </Diario>
        <Diario
          date="2016-10-24"
          key="20161024"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-24"
          />
        </Diario>
        <Diario
          date="2016-10-25"
          key="20161025"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-25"
          />
        </Diario>
        <Diario
          date="2016-10-26"
          key="20161026"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-26"
          />
        </Diario>
        <Diario
          date="2016-10-27"
          key="20161027"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-27"
          />
        </Diario>
        <Diario
          date="2016-10-28"
          key="20161028"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-28"
          />
        </Diario>
        <Diario
          date="2016-10-29"
          key="20161029"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-29"
          />
        </Diario>
      </div>
    </Semanal>
    <Semanal
      ano="2016"
      dias={
        Array [
          30,
          31,
        ]
      }
      key="2016106"
      mes="10"
      periodosSemana={Object {}}
      semana={6}
    >
      <div
        className="semanal"
      >
        <Diario
          date="2016-10-30"
          key="20161030"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-30"
          />
        </Diario>
        <Diario
          date="2016-10-31"
          key="20161031"
          produtividade={0}
        >
          <div
            className="diario c0 "
            title="2016-10-31"
          />
        </Diario>
      </div>
    </Semanal>
  </div>
</Mensal>
`;
